src/DependencyInjection/Configuration.php

```diff
@@ -0,0 +1,56 @@
+<?php
+declare(strict_types=1);
+
+namespace ApiPlatform\PDGBundle\DependencyInjection;
+
+use Symfony\Component\Config\Definition\Builder\TreeBuilder;
+use Symfony\Component\Config\Definition\ConfigurationInterface;
+
+class Configuration implements ConfigurationInterface
+{
+
+    public function getConfigTreeBuilder(): TreeBuilder
+    {
+        $treeBuilder = new TreeBuilder('api_platform_pdg');
+        $rootNode = $treeBuilder->getRootNode();
+
+        $rootNode->children()
+            ->arrayNode('reference')
+                ->children()
+                    ->scalarNode('src')
+                        ->defaultValue('../src')
+                        ->end()
+                    ->arrayNode('patterns')
+                        ->children()
+                            ->arrayNode('directories')
+                                ->scalarPrototype()->end()
+                            ->end()
+                            ->arrayNode('names')
+                                ->scalarPrototype()->end()
+                            ->end()
+                            ->arrayNode('exclude')
+                                ->scalarPrototype()->end()
+                            ->end()
+                            ->arrayNode('class_tags_to_ignore')
+                                ->scalarPrototype()
+                            ->end()
+                    ->end()
+                ->end()
+            ->arrayNode('sidebar')
+                ->children()
+                    ->arrayNode('directories')
+                        ->children()
+                            ->arrayNode('Guides')
+                                ->scalarPrototype()->end()
+                            ->end()
+                            ->arrayNode('Reference')
+                                ->scalarPrototype()->end()
+                            ->end()
```

Comment:
mhh I understand, there's probably something to be done here we'll see that later